variables.tf

```diff
@@ -1,9 +1,33 @@
-##############################################################################
+########################################################################################################################
 # Input Variables
-##############################################################################
+########################################################################################################################
 
-#variable "my_variable" {
-#  type        = string
-#  description = "A description of my variable"
-#  default     = "default_value"
-#}
+variable "enterprise_crn" {
+  type        = string
+  description = "The CRN of the parent Enterprise account to use."
+}
+
+variable "enterprise_primary_contact_iam_id" {
+  type        = string
+  description = "The IAM id of the parent Enterprise account owner."
+}
+
+variable "enterprise_json_input" {
+  description = "List of account groups and account names"
+  type = map(list(object({
+    name         = string
+    owner_iam_id = optional(string, "root")
+    accounts     = optional(list(object({ name = string, owner_iam_id = optional(string, "root") })))
+    account_groups = optional(list(object(
+      { name     = string,
+        accounts = optional(list(object({ name = string, owner_iam_id = optional(string, "root") })))
+        account_groups = optional(list(object({
+          name     = string,
+          accounts = optional(list(object({ name = string, owner_iam_id = optional(string, "root") })))
+          account_groups = optional(list(object(
+          { name = string })))
```

Comment:
I think this depth should also support creating an account too. It doesn't make sense that the module can create this group but not create any account in it. I actually updated advanced example so it would create an account at this level and noticed it was missing, so hence thats hot I found this issue.

Reply:
fixed it